tool/tsh/common/tsh_test.go

```diff
@@ -7767,3 +7767,69 @@ func TestDebugVersionOutput(t *testing.T) {
 		require.Contains(t, string(output), v)
 	}
 }
+
+func TestLogoutOneIdentity(t *testing.T) {
+	tmpHomePath := t.TempDir()
+	connector := mockConnector(t)
+
+	alice, err := types.NewUser("alice@example.com")
+	require.NoError(t, err)
+	alice.SetRoles([]string{"access"})
+
+	rootServer, err := testserver.NewTeleportProcess(
+		t.TempDir(),
+		testserver.WithBootstrap(connector, alice))
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		require.NoError(t, rootServer.Close())
+		require.NoError(t, rootServer.Wait())
+	})
+
+	authServer := rootServer.GetAuthServer()
+	require.NotNil(t, authServer)
+	proxyAddr, err := rootServer.ProxyWebAddr()
+	require.NoError(t, err)
+
+	err = Run(context.Background(), []string{
+		"login",
+		"--insecure",
+		"--proxy", proxyAddr.String()},
+		setHomePath(tmpHomePath),
+		setMockSSOLogin(authServer, alice, connector.GetName()))
+	require.NoError(t, err)
+
+	// logout with one identity with --proxy flag set
+	buf := bytes.NewBuffer([]byte{})
+	err = Run(context.Background(), []string{
+		"logout",
+		"--proxy", proxyAddr.String()},
+		setHomePath(tmpHomePath),
+		func(cf *CLIConf) error {
+			cf.OverrideStdout = buf
+			return nil
+		})
+	require.NoError(t, err)
+	require.Contains(t, buf.String(), fmt.Sprintf("Logged out %v from %v.\n", alice.GetName(), proxyAddr.Host()))
+
+	err = Run(context.Background(), []string{
+		"login",
+		"--insecure",
+		"--proxy", proxyAddr.String()},
+		setHomePath(tmpHomePath),
+		setMockSSOLogin(authServer, alice, connector.GetName()))
+	require.NoError(t, err)
+
+	// logout with TELEPORT_PROXY set
+	t.Setenv(proxyEnvVar, proxyAddr.String())
```

Comment:
This is okay, because you only run one more `tsh login` test after this point.

It is a little bit error prone though, because this env var now remains set for the remainder of the test. This means if you add additional `Run` calls later they'll be impacted.

You could fix this by either:

- Separating the test into two: `TestLogoutOneIdentityProxyFlag` and `TestLogoutOneIdentityProxyEnvVar`, or
- using subtests